packages/nimble-components/src/table/specs/table-spec.md

```diff
@@ -0,0 +1,242 @@
+
+[Table visual design](https://xd.adobe.com/view/33ffad4a-eb2c-4241-b8c5-ebfff1faf6f6-66ac/screen/b9cee5e2-49a4-425a-9ed4-38b23ba2e313/)
+
+### Non-goals
+
+-   The `nimble-table` will not offer spreadsheet-like features such as infinite columns, or summary functions.
+
+-   We do not intend for the `nimble-table` to handle data sets of 50K or more on the client-side. Visualizing such sizes of data sets are meant to be enabled through various "data virtualiazation" means, such as the server only sending over subsets of the data at a time, appropriately filtered down to a reasonable size.
+
+-   Non-virtualized static tables.
+
+### Features
+
+In addition to simply rendering data, the `nimble-table` will provide APIs/interactions that will allow a user to do the following:
+
+-   Interactively and programmatically size columns
+-   Interactively and programmatically sort columns
+-   Pin columns
+-   Provide column ordering (programmatically only at first possibly)
+-   Group rows
+-   Multi-select rows (interactively only at first possibly)
+-   Provide hierarchy via the data
+-   Provide custom column rendering
+-   Provide API for adding action menu
+
+#### _Performance_
+
+We intend for the `nimble-table` to handle both the rendering and interactive operations, such as sorting, in a near-instantaneous fashion on datasets of at least 10K rows. Between 10K and 100K rows of data on the client-side, however, we expect to see a notable drop in performance.
+
+Clients, at least initially, will be expected to create mechanisms to restrict the amount of data coming to the client-side.
+
+#### _Out of Scope of Initial Release_
+
+The following are features that we intend to prioritize eventually after the initial release:
+
+-   Hierarchy and Grouping together
+-   Custom expanded row content
+    -   An example would be showing a sub-tree when you expand a row
+-   Editable cells
+-   APIs around pagination/infinite scrolling
+-   High-level configuration APIs
+    -   Allow for easy getting/setting of total configurable state, which is useful for maintaining state of the table as a user navigates to other pages in an app.
+-   Filter on any text match across all cells
+-   Selection configuration.
+-   Custom header templates
+-   Custom row templates
+
+### Accessibility
+
+The table will provide the necessary means for keyboard interactions and navigation through the cells, as well as expected ARIA labels/roles for the various elements.
+
+### Risks and Challenges
+
+We will be leveraging a couple of third-party libraries, namely [TanStack Table](https://tanstack.com/table/v8) and [TanStack Virtual](https://tanstack.com/virtual/v3), to manage much of the complexity surrounding the table implementation. This presents risk in the form of being able to quickly address any issues that arise. However, these libraries are active and eagerly support outside contribution, so we should be able to contribute as needed.
```

Comment:
> eagerly support outside contribution

Do we actually know this is true? We haven't tried to upstream anything to these libraries. Are you basing this statement on PR responsiveness that you've seen in the repos?

Reply:
Yes, from the [PRs that have been closed recently](https://github.com/TanStack/table/pulls?q=is%3Apr+is%3Aclosed), it appears that there is a decent amount of contribution from a variety of people.